rfcs/0054-suspend-mv-on-non-recoverable-errors.md

```diff
@@ -0,0 +1,63 @@
+---
+feature: Suspend MV on Non-Recoverable Errors
+authors:
+  - "Patrick Huang"
+start_date: "2023/2/24"
+---
+# RFC: Suspend MV on Non-Recoverable Errors
+
+# Motivation
+
+Currently we have two error handling mechanisms on MV:
+
+1. Ignore and tolerate the compute error. Relevant issue: https://github.com/risingwavelabs/risingwave/issues/4625)
+2. Report error to meta and trigger a full recovery.
+
+However, when there are non-recoverable errors caused by malform records ingested by the user, the current error handling mechanisms are insufficient:
+
+1. Even if these messages can be skip or fixed via filling in default values, correctness can be affected, especially when sinking to external systems, since the damage may hardly get reverted.
```

Comment:
> the damage may hardly get reverted.

I'm not completely sure this is correct. 

Could you provide some examples where the damage is permanent? One way iss that their downstream components to RW are affected, and a lot of manual work is required to clean up the wrong data.

For RW itself, I think we should handle by deleting the old MV and creating a new one.

Ultimately, I think the behaviour should be determined by the user, whether they are willing to tolerate certain types of errors or if they want to suspend the job. The same goes for batch queries.

Reply:
> One way iss that their downstream components to RW are affected, and a lot of manual work is required to clean up the wrong data.

Exactly.

> Ultimately, I think the behaviour should be determined by the user, whether they are willing to tolerate certain types of errors or if they want to suspend the job. The same goes for batch queries.

Totally agree. That was my intention too.